Frontend/ui-driver/src/Screens/ChooseCityScreen/View.purs

```diff
@@ -24,29 +23,35 @@ import JBridge as JB
 import Language.Strings (getString)
 import Language.Types (STR(..))
 import MerchantConfig.Types (CityConfig)
-import Prelude (Unit, bind, const, discard, map, not, pure, unit, ($), (<<<), (<>), (==), (&&))
+import Prelude (Unit, bind, const, discard, map, not, pure, unit, ($), (<<<), (<>), (==), (&&), when, void)
 import PrestoDOM (Accessiblity(..), Gradient(..), Gravity(..), Length(..), Margin(..), Orientation(..), Padding(..), PrestoDOM, Prop, Screen, Visibility(..), accessibility, afterRender, alignParentBottom, alpha, background, color, cornerRadius, fontStyle, gradient, gravity, height, id, imageUrl, imageView, imageWithFallback, layoutGravity, linearLayout, margin, onBackPressed, onClick, orientation, padding, relativeLayout, stroke, text, textSize, textView, visibility, weight, width)
 import PrestoDOM.Animation as PrestoAnim
 import Screens.ChooseCityScreen.Controller (Action(..), ScreenOutput, eval)
 import Screens.Types (ChooseCityScreenStage(..), ChooseCityScreenState)
 import Storage (getValueToLocalStore, KeyStore(..))
 import Styles.Colors as Color
+import MerchantConfig.Utils as MU
+import PrestoDOM.Properties as PP
+import PrestoDOM.Types.DomAttributes as PTD
+import Components.ErrorModal as ErrorModal
 
 screen :: ChooseCityScreenState -> Screen Action ChooseCityScreenState ScreenOutput
 screen initialState =
   { initialState
   , view
   , name: "ChooseCity"
   , globalEvents:  [(\ push -> do    
-    if not initialState.props.isLocationPermissionGiven then do
-      _ <- JB.storeCallBackDriverLocationPermission push LocationPermissionCallBack
-      pure unit 
+    if not initialState.props.isLocationPermissionGiven then do 
+      JB.storeCallBackDriverLocationPermission push LocationPermissionCallBack
+      pure unit
     else pure unit
+    when (initialState.data.config.enableMockLocation) $ 
```

Comment:
these when function are working here? did you confirmed it? @ashkriti

Reply:
yes it is working fine

Reply:
Last time when I checked this config was not working